File: SortingTechniques/BubbleSort.c

#include <stdio.h>

void swap(int *a , int *b){
    int temp = *a;
    *a = *b; *b = temp;
}

void bubbleSort(int arr[], int n){
    int i,j,flag;
    for( i = 0; i < n-1; i++){
        flag = 0;
        for( j = 0; j < n-i-1; j++){
            if(arr[j]>arr[j+1]){
                swap(&arr[j],&arr[j+1]);
                flag = 1;
            }
        }
        if(flag==1){
            break;
        }
    }
}

void Kthlargestnum(int arr[], int n, int k){
    int i,j,flag;
    for( i = 0; i < k; i++){
        flag = 0;
        for( j = 0; j < n-i-1; j++){
            if(arr[j]>arr[j+1]){
                swap(&arr[j],&arr[j+1]);
                flag = 1;
            }
        }
        if(flag==1){
            break;
        }
    }
    printf("\n%d",arr[n-k]);
}

int main()
{
    int arr[] = {8,5,7,3,2};
    bubbleSort(arr, 5);
    for(int k = 0; k < 5; k++){
        printf("%d ",arr[k]);
    }
    Kthlargestnum(arr, 5 ,2);
}

File: SortingTechniques/BubbleSort/BubbleSort.java
package SortingTechniques.BubbleSort;

import java.util.Arrays;

public class BubbleSort {
    public static void main(String args[]){
        int[] arr = {3, 2, 5, 7, 1, 8, 9};
        bubbleSort(arr);
        System.out.println(Arrays.toString(arr));
    }

    public static void bubbleSort(int arr[]){
        int n = arr.length;
        for(int i = 0; i < n - 1; i++){
            boolean isSorted = true;
            for(int j = 0; j < n - 1 - i; j++){
                if(arr[j]>arr[j+1]){
                    int temp = arr[j];
                    arr[j] = arr[j+1];
                    arr[j+1] = temp;
                    // swapping happened
                    isSorted = false;
                }
            }
            if(isSorted){
                break;
            }
        }
    }
}
